src/normalize-config.util.ts

```diff
@@ -29,7 +29,9 @@ export function normalizeConfig(config: Config): void {
  */
 export function cleanupConfigForLogging(config: Config): Record<string, any> {
   const configCopy = cloneDeep(config)
```

Comment:
Just a thought, not really a comment to this PR: seeing that we have a function called `cleanupConfigForLogging(...)`, I'd be seriously worried that somebody in future sees that we log configs and logs it themselves, but forgets to put it through the function first.

I guess, I'd prefer to have a custom logging function that we'd use everywhere which would make sure that all the sensitive data is hidden, before putting in logs.

Reply:
I'd set the probability of anyone misusing logging for this to be close to zero, as it requires pretty deep penetration inside the codebase.